include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -44,33 +46,169 @@ namespace engine
 namespace datafacade
 {
 
+template <typename AlgorithmT> class ContiguousInternalMemoryAlgorithmDataFacade;
+
+template <>
+class ContiguousInternalMemoryAlgorithmDataFacade<algorithm::CH>
+    : public datafacade::AlgorithmDataFacade<algorithm::CH>
```

Comment:
Yes. Honestly there is no point in keeping the abstract interfaces anymore. I will remove them in a different PR later.